instructions.json

```diff
@@ -1,111 +1,381 @@
+            },
+            "left": {
+                "default": "At the roundabout turn left",
+                "name": "At the roundabout turn left onto {way_name}",
+                "destination": "At the roundabout turn left towards {destination}"
+            },
+            "right": {
+                "default": "At the roundabout turn right",
+                "name": "At the roundabout turn right onto {way_name}",
+                "destination": "At the roundabout turn right towards {destination}"
+            },
+            "straight": {
+                "default": "At the roundabout continue straight"
+            }
         },
         "train": {
-            "defaultInstruction": "Take the train[ - {way_name}]"
+            "default": {
+                "default": "Take the train",
+                "name": "Take the train - {way_name}"
+            }
         },
         "turn": {
-            "defaultInstruction": "Make a {modifier}[ onto {way_name}]",
-            "left": "Turn left[ onto {way_name}]",
-            "right": "Turn right[ onto {way_name}]",
-            "straight": "Go straight[ onto {way_name}]"
+            "default": {
+                "default": "Make a {modifier}",
+                "name": "Make a {modifier} onto {way_name}",
+                "destination": "Make a {modifier} towards {destination}"
+            },
+            "left": {
+                "default": "Turn left",
+                "name": "Turn left onto {way_name}",
+                "destination": "Turn left towards {destination}"
+            },
+            "right": {
+                "default": "Turn right",
+                "name": "Turn right onto {way_name}",
+                "destination": "Turn right towards {destination}"
+            },
+            "straight": {
+                "default": "Go straight",
+                "name": "Go straight onto {way_name}",
+                "destination": "Go straight towards {destination}"
+            }
         },
         "use lane": {
-            "defaultInstruction": "Continue straight[ onto {way_name}]",
-            "left": "{laneInstruction} to turn left[ onto {way_name}]",
-            "right": "{laneInstruction} to turn right[ onto {way_name}]",
-            "sharp left": "{laneInstruction} to make a sharp left[ onto {way_name}]",
-            "sharp right": "{laneInstruction} to make a sharp right[ onto {way_name}]",
-            "slight left": "{laneInstruction} to make a slight left[ onto {way_name}]",
-            "slight right": "{laneInstruction} to make a slight right[ onto {way_name}]",
-            "uturn": "{laneInstruction} to make a U-turn[ onto {way_name}]",
+            "default": {
+                "default": "Continue straight",
+                "name": "Continue straight onto {way_name}",
+                "destination": "Continue straight towards {destination}"
+            },
+            "left": {
+                "default": "{laneInstruction} to turn left",
```

Comment:
For the purpose of minimizing translation cost, it may be desirable at some point to refactor this so that there would literally be a string `{laneInstruction} to {instruction}`. I'm not aware of any languages that would need more granularity in this case, but I could be wrong.